core/src/smartcontracts/wasm.rs

```diff
@@ -191,6 +202,8 @@ struct State<'wrld> {
     triggering_event: Option<Event>,
     /// Operation to pass to a runtime permission validator
     operation_to_validate: Option<permission::validator::NeedsPermissionBox>,
```

Comment:
Because we still need to pass three variants:

1. `SignedTransaction` -- then we want to check incoming transaction
2. `Instruction` -- then we want to check instruction executed by some wasm
3. `Query` -- then we want to check incoming query

Reply:
it's not possible for `operation_to_validate` to be `SignedInstruction`?

Reply:
Maybe you mean `SignedTransaction`? If so, then it's still not possible to pass queries, as they aren't transactions and still not possible to pass instructions executed by smart contracts

Reply:
yes, it's what I mean. This field can never hold `NeedsPermissionBox::Transaction` variant. Would that mean we don't need that variant at all?

Reply:
Why? We use `NeedPermissionBox::Transaction` to validate every transaction for users